### DISCUSSION TITLE
How can I prevent the button from getting smaller when pressed?

### DISCUSSION THREAD

**Discussion by Irasil** [AvaloniaUI/Avalonia#16100] [Q&A] [answered]
I've tried using the `:pressed` pseudo-class. The background color turns red, but the button doesn't get larger; instead, it becomes slightly smaller while pressed, just like the default behavior. What I want to achieve is that the size of the button does not change when pressed.
This is probably a very simple issue, but I can't find a solution.

What I tried:

```csharp
<Window.Styles>
  <Style Selector="Button.mainButton /template/ ContentPresenter">
      <Setter Property="Background" Value="#4798d8"/>
      <Setter Property="BorderBrush" Value="Transparent"></Setter>
          <Style Selector="^:pointerover /template/ ContentPresenter">
              <Setter Property="Background" Value="#4798d8"/>
              <Setter Property="BorderBrush" Value="#4798d8"></Setter>
          </Style>			    
  </Style>
  
  <Style Selector="Button.mainButton:pressed /template/ ContentPresenter">
          <Setter Property="Width" Value="600"/>
	  <Setter Property="Height" Value="250"/>
	  <Setter Property="Background" Value="red"></Setter>
  </Style>		
</Window.Styles>

<Grid>
<Button Name="MainButton"
              Width="500" Height="150"
              Cursor="Hand"
              Focusable="False"
		Classes="mainButton">
```

**Comment by stevemonaco** [comment, marked as answer]
First, you always need to read the original `ControlTheme`. For Fluent, the relevant bit is here:

https://github.com/AvaloniaUI/Avalonia/blob/2a9e7e9dcfc3e590bdd50b1e874ce70c0fafcc24/src/Avalonia.Themes.Fluent/Controls/Button.xaml#L51-L53

In this instance, you can easily replace this with a new selector:

```xml
<Style Selector="Button:pressed">
  <Setter Property="RenderTransform" Value="{x:Null}" />
</Style>
```

You could also set `Value="scale(1)"`, but no transform should be more efficient.
